app/javascript/controllers/application_controller.js

```diff
@@ -0,0 +1,63 @@
+"use strict"
+
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["newMessageField"]
+
+  connect() {
+    document.addEventListener("keydown", this.keypress.bind(this), false)
+  }
+
+  disconnect() {
+    document.removeEventListener("keydown", this.keypress.bind(this))
+  }
+
+  keypress(event) {
+    let matched = false
+
+    switch(event.key) {
+      // Shift+Esc focuses chat input
+      case "Escape":
+        if (event.shiftKey) {
+          this.newMessageFieldTarget.focus()
+          matched = true
+        }
+        break
+
+      // `/` focuses the message input, if not already focused
+      case "/":
+        if ( ! this.newMessageFieldTarget.matches(":focus")) {
+          this.newMessageFieldTarget.focus()
+          matched = true
+        }
+        break
+
+      // Cmd+Shift+S toggles the sidebar
+      case "s":
+        if (event.metaKey && event.shiftKey) {
```

Comment:
This is _probably_ the wrong shortcut for windows and linux users, which would prefer the ctrl key here.